### DISCUSSION TITLE
C2DGraphics and serial ports

### DISCUSSION THREAD

**Discussion by KyleCardoza** [rsta2/circle#503] [Q&A] [answered]
I feel as if I'm missing something blindingly obvious: when I try to make a little program to show a square on the screen and output some text on the flashy serial monitor, no matter what I try, it just sits at a black screen. 

If I remove the serial and interrupt devices from the code, then it draws to the screen no problem. 

So I guess my question is, if I want to use C2DGraphics, am I just out of luck for IRQs and debug serial output, or am I missing something when I study the source of the sample programs?

**Comment by rsta2** [comment, marked as answer]
You have to add a member of `CDeviceNameService` to `CKernel` before `CSerialDevice`, because the serial device wants to create a device name `ttyS1` in the service. An instance of `CInterruptSystem` is not necessary any more in `CKernel` since Circle 46, because it is already created in sysinit, but it does also not disturb (for backwards compatibility).

**Reply by KyleCardoza** [reply]
Thank you very much. I have added that, and on its own it works fine, but when I then add in C2DGraphics, it freezes before displaying anything -- it works fine without C2DGraphics, happily sending strings over the serial monitor.

**Reply by KyleCardoza** [reply]
Never mind. I figured it out -- I forgot the logger. As you've said, no program of any complexity will work without the logger. Don't know what I was thinking. Anyway, thanks, love your work.

**Reply by rsta2** [reply]
It's good, that it works now. Thanks for appreciating Circle!
